packages/updating-element/src/updating-element.ts

```diff
@@ -789,56 +589,38 @@ export abstract class UpdatingElement extends HTMLElement {
     }
   }
 
-  /**
-   * Requests an update which is processed asynchronously. This should be called
-   * when an element should update based on some state not triggered by setting
-   * a reactive property. In this case, pass no arguments. It should also be
-   * called when manually implementing a property setter. In this case, pass the
-   * property `name` and `oldValue` to ensure that any configured property
-   * options are honored.
-   *
-   * @param name {PropertyKey} (optional) name of requesting property
-   * @param oldValue {any} (optional) old value of requesting property
-   * @param options {PropertyDeclaration} (optional) property options to use
-   * instead of the previously configured options
-   */
+  /** @internal */
+  _propertyChanged(
+    name: PropertyKey,
+    oldValue: unknown,
+    options: PropertyDeclaration
+  ) {
+    super._propertyChanged(name, oldValue, options);
+    // Add to reflecting properties set.
+    // Note, it's important that every change has a chance to add the
+    // property to `_reflectingProperties`. This ensures setting
+    // attribute + property reflects correctly.
+    if (options.reflect === true && this._reflectingProperty !== name) {
+      if (this._reflectingProperties === undefined) {
+        this._reflectingProperties = new Map();
+      }
+      this._reflectingProperties.set(name, options);
+    }
+  }
+
   requestUpdate(
     name?: PropertyKey,
     oldValue?: unknown,
     options?: PropertyDeclaration
   ) {
-    let shouldRequestUpdate = true;
-    // If we have a property key, perform property update steps.
-    if (name !== undefined) {
-      options =
-        options ||
-        (this.constructor as typeof UpdatingElement).getPropertyOptions(name);
-      const hasChanged = options.hasChanged || notEqual;
-      if (hasChanged(this[name as keyof this], oldValue)) {
-        if (!this._changedProperties.has(name)) {
-          this._changedProperties.set(name, oldValue);
-        }
-        // Add to reflecting properties set.
-        // Note, it's important that every change has a chance to add the
-        // property to `_reflectingProperties`. This ensures setting
-        // attribute + property reflects correctly.
-        if (options.reflect === true && this._reflectingProperty !== name) {
-          if (this._reflectingProperties === undefined) {
-            this._reflectingProperties = new Map();
-          }
-          this._reflectingProperties.set(name, options);
-        }
-      } else {
-        // Abort the request if the property should not be considered changed.
-        shouldRequestUpdate = false;
-      }
-    }
-    if (!this.isUpdatePending && shouldRequestUpdate) {
+    const needsUpdate =
+      super.requestUpdate(name, oldValue, options) && !this.isUpdatePending;
```

Comment:
What is the significance of the `isUpdatePending` flag being implemented/checked here in UpdatingElement and not in UpdatingMixin?

Reply:
`UpdatingMixin` has the updating lifecycle interface but it doesn't implement it so it doesn't have any notion of `isUpdatePending`. So this means trigger an update if the change suggested we should and one is not already pending.